internal/clients/gitdiff/gitdiff.go

```diff
@@ -0,0 +1,88 @@
+package gitdiff
+
+import (
+	"bytes"
+
+	"github.com/nightfallai/jenkins_test/internal/clients/diffreviewer"
+	"github.com/nightfallai/jenkins_test/internal/interfaces/libgitintf"
+)
+
+// Client is a wrapper around the libgit interface
+type Client struct {
+	Libgit       libgitintf.Libgit
+	RepoFilePath string
+}
+
+// DiffOptions options specifying the file diffs to be returned
+type DiffOptions struct {
+	FilterLineType map[diffreviewer.LineType]bool
+}
+
+// NewClient creates a libgit client
+func NewClient(accessToken, repoFilePath string) *Client {
+	return &Client{
+		Libgit:       NewLibgit(accessToken),
+		RepoFilePath: repoFilePath,
+	}
+}
+
+// GetDiff gets the diff from the base to the head on the given repo
+func (c *Client) GetDiff(baseRev, headRev, repoURL string, diffOpts *DiffOptions) ([]*diffreviewer.FileDiff, error) {
+	repo, err := c.Libgit.Clone(repoURL, c.RepoFilePath)
```

Comment:
what happens here for giant repos?

Reply:
Since the clone also downloads all the history of the repo for large repos it takes a while. I am investigating using a shallow repo and fetching the commits we need to compute the diff.